### PR TITLE
(DOCSP-29241): Bulk Usage Example page

### PR DESCRIPTION
**Pull request by cbullinger**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-java/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-29241
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/kotlin/docsworker-xlarge/docsp-29241-bulk-usage/usage-examples/bulkWrite/

## Updated Page

Usage Examples > [Perform Bulk Operations page](https://www.mongodb.com/docs/drivers/java/sync/current/usage-examples/bulkWrite/)

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Did you run a spell-check?
- [ ] Did you run a grammar-check?
- [ ] Are all the links working?

### CODE REVIEW

**Review by mongodben — APPROVED** [review]
lgtm w 1 tiny suggestion

File: examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
+package usageExamples.bulkWrite
+
+// :replace-start: {
+//    "terms": {
+//       "CONNECTION_URI_PLACEHOLDER": "\"<connection string uri>\"",
+//       "import io.github.cdimascio.dotenv.dotenv\n": ""
```
Comment (rozza) [inline examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt:6]: TIL about dotenv - seems neat :)

File: examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+//    }
+// }
+// :snippet-start: bulk-write-usage-example
+import com.mongodb.MongoException
+import com.mongodb.client.model.*
```
Comment (rozza) [inline examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt:11]: import * is generally frowned upon as explicit imports are better than implicit ones. Star imports aren't used in the java examples.

Has it been agreed that its ok for Kotlin docs?

**Review by rozza — COMMENTED** [review]
Just a couple of questions.

Out of interest will the documentation just be for coroutine based kotlin code?

File: examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+import com.mongodb.kotlin.client.coroutine.MongoClient
+import io.github.cdimascio.dotenv.dotenv
```
[the remark this answers is not in the document]
Comment (rozza) [reply examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt:13]: Nm. I see it gets replaced out.

File: examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt (hunk 2 above)
Comment (cbullinger) [reply examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt:11]: Good catch. It hasn't been discussed, but we've been operating under the assumption that we're matching the Java docs as much as possible. I'll disable IntelliJ's import optimization and will update the code and snippet to use the explicit imports/match the Java example

**Comment by cbullinger** [comment]
@rozza yes, everything [we've documented](https://www.mongodb.com/docs/drivers/kotlin/coroutine/current/) (and everything you'll be reviewing) are for the coroutine driver only. For now, the sync driver is undocumented.

File: examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt (hunk 2 above)
Comment (cbullinger) [reply examples/src/main/kotlin/usageExamples/bulkWrite/BulkWrite.kt]: Updated

**Pull request merged**